shared/defs.proto

```diff
@@ -181,3 +181,9 @@ message Beam {
     float minor_axis = 4;
     float pa = 5;
 }
+
+message Progress {
```

Comment:
Could we give this a more specific name, and also more documentation in the comments? Out of context it's unclear what this message is for.

Reply:
All of these proto file additions should be commented -- the docs are automatically generated from these comments. Any descriptions that are written by hand only into the docs are going to disappear the next time the docs are regenerated.

Reply:
Perhaps `ListProgress` would be an appropriate generic name. I agree with @angus that there should be only one event type.